overlord/snapstate/handlers.go

```diff
@@ -3410,11 +3402,27 @@ func (m *SnapManager) doKillSnapApps(t *state.Task, _ *tomb.Tomb) (err error) {
 	perfTimings := state.TimingsForTask(t)
 	defer perfTimings.Save(st)
 
+	inhibitInfo := runinhibit.InhibitInfo{Previous: snapsup.Revision()}
+	if err := runinhibit.LockWithHint(snapName, runinhibit.HintInhibitedForRemove, inhibitInfo, st.Unlocker()); err != nil {
+		return err
+	}
+
 	// State lock is not needed for killing apps or stopping services and since those
 	// can take some time, let's unlock the state
 	st.Unlock()
 	defer st.Lock()
 
+	// Note: The snap hint lock file is completely removed in “discard-snap”
+	// so we only need to unlock it in case of an error here or during undo.
+	defer func() {
+		// Unlock snap inhibition if anything goes wrong afterwards to
+		// avoid keeping the snap stuck at this inhibited state.
+		if err != nil {
+			// state is unlocked, it is okay to pass nil here
+			runinhibit.Unlock(snapName, nil)
```

Comment:
is this code covered by tests?

Reply:
Yes, This is tested in `TestDoKillSnapAppsUnlocksOnError`

https://github.com/canonical/snapd/blob/8f5a8aa8b53456353c8d42e87df8515546350274/overlord/snapstate/handlers_link_test.go#L2740-L2743